tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/ApplyCondition.cs

```diff
@@ -70,21 +79,33 @@ public ApplyCondition(JsonElement jsonElement)
             }
         }
 
-        // sanitizers run against RecordEntries
         public bool IsApplicable(RecordEntry entry)
         {
-            var result = _regex.Match(entry.RequestUri);
-            return result.Success;
-        }
+            if (UriRegex != null)
+            {
+                if (!Regex.IsMatch(entry.RequestUri, UriRegex))
+                {
+                    return false;
+                }
+            }
 
-        // transforms are used directly on the request
-        public bool IsApplicable(HttpRequest request)
-        {
-            var requestString = request.HttpContext.Features.Get<IHttpRequestFeature>().RawTarget;
-            var requestUri = request.Scheme + "://" + request.Host + requestString;
+            if (ResponseHeaderCondition != null)
+            {
+                if (!entry.Response.Headers.ContainsKey(ResponseHeaderCondition.Key))
+                {
+                    return false;
+                }
+
+                if (ResponseHeaderCondition.ValueRegex != null)
+                {
+                    if (!Regex.IsMatch(entry.Response.Headers[ResponseHeaderCondition.Key].First(), ResponseHeaderCondition.ValueRegex))
```

Comment:
Is ToString() or First() the right option here?

Reply:
Hmm, probably we should return true if at least one of the values matches.